app/views/handlers/feedback.py

```diff
@@ -58,7 +63,27 @@ def get_page_title(self):
         return self.PAGE_TITLE
 
     def handle_post(self):
+        session_data = self._session_store.session_data
         self._session_store.session_data.feedback_count += 1
+
+        data = {
+            "feedback_text": self.form.data["feedback-type"],
+            "feedback_topic": self.form.data["feedback-text"],
```

Comment:
The new `'feedback-type-question-category'` should be conditionally added to `data` if if has a valid value.

Reply:
Its on a different card

Reply:
Okay didn't realise one existed but I also think it is suitable to add here given this it is 2 lines of code and doing a separate card for that and reviews would just be longer. 

However, regardless if the above is done separately, this reminds me that it can't be tested anyway without adding unit test for handler. Therefore, it probably should be a different class that builds the payload like we do with IR which can then be unit tested as at the moment, we have no guarantee that the submitted feedback will arrive in GCS as expected without manual testing.

Reply:
This is a integration test that checks the blobs payload and metadata

Reply:
> There is an integration test that checks the blobs payload and metadata and this is how submitter does it

Sorry I don't see it, where exactly?. If you are referring to the test of GCSFeedback class then that is not the same thing, that is testing that the submitter can submit to GCS, but we need assurances of the feedback payload. For submitter, we introduced a submission handler which has a `get_payload` method which we partially test so this could be similar to that but the payload for feedback has to be tested independently to the submitter.

Reply:
That's not quite true, I added the that test because I couldn't get to the submission_language_code, the rest of submission payload is not tested

Reply:
> That's not quite right, I added that test because I couldn't get to the submission_language_code, the rest of submission payload is not tested. 

Because that is what you added so tested the new bit you introduced. But because we don't have a test to check the submission payload shouldn't be a reason to not have a test for this...

> The integration test creates a GCSFeedback class gives it metadata and the data, then asserts the blob is correct which will be uploaded, this is more than submitter does, I don't follow what more we need

`GCSFeedback` is a submitter, it tests that the submitter works, it has nothing to do with feedback really. It tests that given some data and metadata (which could be anything in any structure you set up in test), the blob is as expected which is not the same as testing that for a feedback submission the object contains the expected properties and values like we do with IR and `ConfirmationEmail`.